### PR TITLE
docs(typescript): instruct to use legacy DT types as a starting point

### PR DESCRIPTION
**Pull request by tay1orjones**
Suggestions from @mbarrer - we want to ensure that the old types provided via DT are used as a starting point for the types we're working to provide directly through the package.

#### Changelog

**Changed**

- update typescript docs

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]

File: docs/guides/adding-component-types.md [hunk 1]
```diff
@@ -92,8 +92,19 @@ the repository. There is an issue tracking curring status of this effort,
...
+- Look for any
+  [legacy types for the component, previously provided via DefinetlyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/carbon-components-react/lib/components)
+  - Copy these legacy types over to your working file and use them as a starting
+    point.
+  - NOTE: If there are legacy types available, these were originally provided
+    for `carbon-components-react@v7.x` and may be out of date.
+  - Even still, it would be best to have as much parity as possible between
+    legacy types for `carbon-components-react@v7.x` and the new typings we're
+    providing for `@carbon/react@1.x`.
+  - After copying them over to your working fileß, ensure the legacy types don't
```
Comment (tw15egan) [inline docs/guides/adding-component-types.md]:
```suggestion
  - After copying them over to your working file, ensure the legacy types don't
```

**Review by mbarrer — APPROVED** [review]

File: docs/guides/adding-component-types.md [hunk 2]
```diff
@@ -92,8 +92,19 @@ the repository. There is an issue tracking curring status of this effort,
 - Change extension to `.tsx` via git (do not simply rename)
   - e.g.
     `git mv packages/react/src/components/ComponentName/ComponentName.js packages/react/src/components/ComponentName/ComponentName.tsx`
-- Copying the proptypes def to above the component definition
+- Copy the proptypes def to above the component definition
 - Retool the proptypes to be a ts interface
+- Look for any
+  [legacy types for the component, previously provided via DefinetlyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/carbon-components-react/lib/components)
```
Comment (metonym) [inline docs/guides/adding-component-types.md]:
```suggestion
  [legacy types for the component, previously provided via [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped)](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/carbon-components-react/lib/components)
```

Comment (tay1orjones) [reply docs/guides/adding-component-types.md]: I don't think we can nest links within one another like this

Comment (metonym) [reply docs/guides/adding-component-types.md]: My mistake – was meaning to highlight the `DefinetlyTyped` typo but I think GitHub's copy/paste added the full link.

```diff
- DefinetlyTyped
+ DefinitelyTyped
```

Comment (tay1orjones) [reply docs/guides/adding-component-types.md]: oy! Yes I always misspell it! Thank you 🙏

**Review by aledavila — APPROVED** [review]

**Pull request merged**